src/main/java/com/openlattice/authorization/EdmAuthorizationHelper.java

```diff
@@ -137,7 +221,7 @@ public EdmAuthorizationHelper( EdmManager edm, AuthorizationManager authz ) {
             EnumSet<Permission> requiredPermissions,
             Set<Principal> principals ) {
         return ( entitySetIds.isEmpty() )
-                ? ImmutableMap.of()
+                ? Maps.newHashMap()
```

Comment:
Does the map have to be mutable?

Reply:
Well, yes, because when collecting authorized properties on multiple entity sets, we collect it for linking entity sets and normal entity sets separately, but after representing it as just 1 map. I think it's faster to just put all elements into the first map, rather to create a new map with the copy of the first map and do the same things after.

Reply:
Makes sense unless the algorithm changes, which I think it should. See general comment.